alacritty/src/display/mod.rs

```diff
@@ -1371,23 +1377,38 @@ impl Display {
     }
 
     /// Requst a new frame for a window on Wayland.
-    #[inline]
-    #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
-    fn request_frame(&self, window: &Window) {
-        let surface = match window.wayland_surface() {
-            Some(surface) => surface,
-            None => return,
-        };
+    fn request_frame(&mut self, scheduler: &mut Scheduler) {
+        // Mark that we've used a frame.
+        self.window.has_frame.store(false, Ordering::Relaxed);
 
-        let should_draw = self.window.should_draw.clone();
+        #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
+        if let Some(surface) = self.window.wayland_surface() {
+            let has_frame = self.window.has_frame.clone();
+            // Request a new frame.
+            surface.frame().quick_assign(move |_, _, _| {
+                has_frame.store(true, Ordering::Relaxed);
+            });
 
-        // Mark that window was drawn.
-        should_draw.store(false, Ordering::Relaxed);
+            return;
+        }
 
-        // Request a new frame.
-        surface.frame().quick_assign(move |_, _, _| {
-            should_draw.store(true, Ordering::Relaxed);
-        });
+        // Get the display vblank interval in micro seconds.
+        let monitor_vblank_interval = Duration::from_micros(
+            (1e6 / (self
```

Comment:
I find the `1e6` notation difficult to grasp at a glance and I rarely see that in source code likely for that reason. Sure it's shorter, but is it more readable than `1_000_000`?

If you go into the billions and higher it might be more helpful, but I think people will recognize much faster that `1_000_000` is a million rather than `1e6`.

This also applies to the `1e3` used later which is even worse since it's just 1000.